code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/legionnaire.dm

```diff
@@ -0,0 +1,293 @@
+	icon_state = "legionnaire_head"
+	icon_living = "legionnaire_head"
+	icon_aggro = "legionnaire_head"
+	icon_dead = "legionnaire_dead"
+	icon_gib = "syndicate_gib"
+	maxHealth = 80
+	health = 80
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	attack_verb_continuous = "bites at"
+	attack_verb_simple = "bite at"
+	attack_sound = 'sound/effects/curse1.ogg'
+	throw_message = "simply misses"
+	speed = 0
+	move_to_delay = 2
+	del_on_death = 1
+	deathmessage = "crumbles away!"
+	faction = list()
+	ranged = 0
+	var/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/body = null
+	
+/mob/living/simple_animal/hostile/asteroid/elite/legionnairehead/death()
+	. = ..()
+	if(body)
+		body.onHeadDeath()
+		
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/head_detach(target)
+	ranged_cooldown = world.time + 5
+	if(has_head)
+		has_head = FALSE
+		icon_state = "legionnaire_headless"
+		icon_living = "legionnaire_headless"
+		icon_aggro = "legionnaire_headless"
+		visible_message("<span class='boldwarning'>[src]'s head flies off!</span>")
+		var/mob/living/simple_animal/hostile/asteroid/elite/legionnairehead/newhead = new /mob/living/simple_animal/hostile/asteroid/elite/legionnairehead(loc)
+		newhead.flags_1 |= (flags_1 & ADMIN_SPAWNED_1)
+		newhead.GiveTarget(target)
+		newhead.faction = faction.Copy()
+		myhead = newhead
+		myhead.body = src
+		if(health < maxHealth * 0.25)
+			myhead.melee_damage_lower = 30
+			myhead.melee_damage_upper = 30
+		else if(health < maxHealth * 0.5)
+			myhead.melee_damage_lower = 20
+			myhead.melee_damage_upper = 20
+		return
+	else if(!has_head && myhead != null)
+		myhead.adjustBruteLoss(600)
+		
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/onHeadDeath()
+	myhead = null
+	addtimer(CALLBACK(src, .proc/regain_head), 50)
+	
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/regain_head()
+	has_head = TRUE
+	if(stat == DEAD)
+		return
+	icon_state = "legionnaire"
+	icon_living = "legionnaire"
+	icon_aggro = "legionnaire"
+	visible_message("<span class='boldwarning'>The top of [src]'s spine leaks a black liquid, forming into a skull!</span>")
+	
+/obj/structure/legionnaire_bonfire
+	name = "bone pile"
+	desc = "A pile of bones which seems to occasionally move a little.  It's probably a good idea to smash them."
+	icon = 'icons/obj/lavaland/legionnaire_bonfire.dmi'
+	icon_state = "bonfire"
+	max_integrity = 100
+	move_resist = MOVE_FORCE_EXTREMELY_STRONG
+	anchored = TRUE
+	density = FALSE
+	light_range = 4
+	light_color = LIGHT_COLOR_RED
+	var/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/myowner = null
+	
+	
+/obj/structure/legionnaire_bonfire/Entered(atom/movable/mover, turf/target)
+	if(isliving(mover))
+		var/mob/living/L = mover
+		L.adjust_fire_stacks(3)
+		L.IgniteMob()
+	. = ..()
+	
+/obj/structure/legionnaire_bonfire/deconstruct(disassembled)
+	if(myowner != null)
+		myowner.mypile = null
```

Comment:
should be in `Destroy` as well, in case it gets deleted without being deconstructed